core/src/parse.rs

```diff
@@ -232,98 +226,115 @@ fn jump_stmt(input: &str) -> ParseResult<'_, Statement> {
     ))(input)
 }
 
-// TODO: for now throwing away spaces and comments
-// probably want too keep them when we do source mapping
-fn comment(input: &str) -> ParseResult<'_, &str> {
-    let (input, _) = preceded(
-        space0,
-        context(
-            "Comment",
-            terminated(char(';'), cut(many_till(anychar, line_ending))),
-        ),
-    )(input)?;
-    Ok((input, ""))
+/// Parses a line comment, which starts with a ; and runs to the end of the
+/// line. This terminates at the line ending, but does _not_ consume it.
+fn line_comment(input: &str) -> ParseResult<'_, &str> {
+    map(
+        context("Comment", preceded(char(';'), many0(is_not("\r\n")))),
+        |_| "", // throw the comment away
+    )(input)
 }
 
-fn comment_or_spaces(input: &str) -> ParseResult<'_, &str> {
-    let (input, _) =
-        many0(tuple((space0, many1(alt((comment, line_ending))), space0)))(
-            input,
-        )?;
-    Ok((input, ""))
+/// Parses one source statement. This consumes surrounding spaces/comments, but
+/// not the line ending.
+fn statement(input: &str) -> ParseResult<'_, Statement> {
+    context("Statement", alt((label_stmt, operator_stmt, jump_stmt)))(input)
```

Comment:
I think the comment doesn't match. Doesn't look like the parsers in the alt consume surrounding spaces/comments